### ISSUE TITLE
Compress-Archive creates zero files duplicating directories

### ISSUE THREAD

**Issue by dermeister0** [PowerShell/Microsoft.PowerShell.Archive#27]
Steps to reproduce
------------------

Execute a test script.

```powershell
if (Test-Path 'Root')
{
    Remove-Item 'Root' -Recurse -Force
}
Remove-Item *.zip

$path = 'Root\Level0\LastDir'
New-Item -ItemType Directory $path | Out-Null
New-Item -ItemType Directory "$path\EmptyDir" | Out-Null
Set-Content "$path\Document.txt" (Get-Date)

Compress-Archive 'Root' 'Test1.zip'

Add-Type -AssemblyName 'System.IO.Compression.FileSystem'
[System.IO.Compression.ZipFile]::CreateFromDirectory("$PSScriptRoot\Root", "$PSScriptRoot\Test2.zip")

$sevenZip = 'C:\Program Files\7-Zip\7z.exe'
if (Test-Path $sevenZip)
{
    &$sevenZip a 'Test3.zip' 'Root'
}
```

Test1.zip is created by PowerShell, Test2.zip - by .NET 4.5 ZipFile, Test3.zip - by 7-Zip.

Test1.zip can't be extracted.

![2017-02-02_2210_001](https://cloud.githubusercontent.com/assets/4054844/22555451/52dcf068-e996-11e6-8ae8-232bd5c7348f.png)

![2017-02-02_2212](https://cloud.githubusercontent.com/assets/4054844/22555460/589c0822-e996-11e6-9d62-9a3b7ae4478c.png)

Test2.zip has invalid zero file instead of empty directory.

![2017-02-02_2206](https://cloud.githubusercontent.com/assets/4054844/22555434/3eb3ab86-e996-11e6-8588-5aa41298cba5.png)

Test3.zip is correct.

![2017-02-02_2210](https://cloud.githubusercontent.com/assets/4054844/22555482/677eefa8-e996-11e6-80c9-b5a8454d710b.png)

Expected behavior
-----------------

Created archive with correct structure.

```
Root/Level0/LastDir/EmptyDir/
Root/Level0/LastDir/Document.txt
```

Actual behavior
---------------

Every directory that does not contain files has a duplicating zero-length file in the archive.
Empty directory is converted to file.

```
Root/Level0/LastDir/Document.txt
Root/Level0/LastDir/EmptyDir
Root/Level0
```

![2017-02-02_2152](https://cloud.githubusercontent.com/assets/4054844/22555421/348cea64-e996-11e6-917f-9b2161e2f0bc.png)

![2017-02-02_2153](https://cloud.githubusercontent.com/assets/4054844/22555427/370b9d58-e996-11e6-9bff-4c0bf156e9ad.png)

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.14393.693
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.693
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by dermeister0** [comment]
The issue persists in gallery too.

> Microsoft.PowerShell.Archive 1.0.1.0

**Comment by paulirwin** [comment]
I can reproduce this on PowerShell 5 as well as 6.1 (Core). Workaround is to put a dummy file in each folder, but would really like to see this get fixed.

**Comment by ayousuf23** [comment]
This is fixed in 2.0.1.
